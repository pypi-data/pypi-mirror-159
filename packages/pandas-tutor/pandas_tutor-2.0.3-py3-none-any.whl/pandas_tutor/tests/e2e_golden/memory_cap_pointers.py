# tests that pointers still trigger memory cap


class MyStr:
    def __init__(self, val):
        self.val = val


# this uses about 2.5 MB
# but, sys.getsizeof() only returns ~120 KB
hello = [MyStr('hello world') for _ in range(15_000)]

hello[:5]
