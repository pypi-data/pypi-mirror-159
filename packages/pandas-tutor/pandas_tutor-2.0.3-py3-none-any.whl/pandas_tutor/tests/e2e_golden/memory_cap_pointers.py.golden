{
  "code": "# tests that pointers still trigger memory cap\n\n\nclass MyStr:\n    def __init__(self, val):\n        self.val = val\n\n\n# this uses about 2.5 MB\n# but, sys.getsizeof() only returns ~120 KB\nhello = [MyStr('hello world') for _ in range(15_000)]\n\nhello[:5]\n",
  "explanation": [
    {
      "type": "RuntimeErrorInChain",
      "code_step": "hello[:5]",
      "message": "Your total data uses 2.29 MB of memory, which exceeds the maximum of 1.00 MB that this tool supports.",
      "fragment": {
        "start": {
          "line": 0,
          "ch": 5
        },
        "end": {
          "line": 0,
          "ch": 9
        }
      }
    }
  ]
}